source_code/main_mcu/src/COMMS/comms_aux_mcu.c

```diff
@@ -243,7 +244,112 @@ void comms_aux_mcu_deal_with_received_event(aux_mcu_message_t* received_message)
             break;
         }
         default: break;
-    }    
+    }
+}
+
+/*! \fn     comms_aux_mcu_handle_FIDO2_auth_cred_msg(FIDO2_message_t const *received_message, aux_mcu_message_t *send_message)
+*   \brief  routine handling authenticating a credential
+*   \param  received_message The received message
+*   \param  send_message The response message
+*   \return FIDO2_MSG_RCVD
+*/
+static comms_msg_rcvd_te comms_aux_mcu_handle_FIDO2_auth_cred_msg(FIDO2_message_t const *received_message, aux_mcu_message_t *send_message)
+{
+    FIDO2_auth_cred_req_message_t const *incoming_message = &received_message->FIDO2_auth_cred_req_message;
+    FIDO2_auth_cred_rsp_message_t *response = &send_message->FIDO2_message.FIDO2_auth_cred_rsp_message;
+
+    fido2_process_exclude_list_item(incoming_message, response);
+
+    /* Set same message type and payload length. We are using the same message
+     * in both directions
+     */
+    send_message->message_type = AUX_MCU_MSG_TYPE_FIDO2;
+    send_message->FIDO2_message.message_type = AUX_MCU_MSG_TYPE_AUTH_CRED_RSP;
+    send_message->payload_length1 = sizeof(FIDO2_message_t);
```

Comment:
here you're returning the size of the message_type + reserved + the union. Are you sure the receiving end on the AUX MCU can deal with the non used bytes?

Reply:
I am not quite sure what you mean. Both AUX_MCU and MAIN_MCU use the same structure to access the data. All the data is initialized (to 0) before structure is filled. I don't know what you mean by dealing with "unused" bytes....

Reply:
well the payload length field is supposed to let the receiving party how many useful bytes there are... so here you're letting the receiving party know that there are more bytes than there actually is. But given you use the enum on the receiving end, I guess it's ok :)